dev-lang/rust/rust_bin-1.21.0.recipe

```diff
@@ -0,0 +1,94 @@
+SUMMARY="Modern and safe systems programming language"
+DESCRIPTION="Rust is a systems programming language that runs blazingly fast, prevents almost all crashes*, and eliminates data races."
+HOMEPAGE="http://www.rust-lang.org/"
+LICENSE="MIT"
+COPYRIGHT="2017, The Rust Project Developers"
+REVISION="1"
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+case "$effectiveTargetArchitecture" in
+	x86)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-i686-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="465db013876ead01686c991f1fa63ef4a6687d3cabbcde81f5a756e367989cdf"
+SOURCE_DIR="rust-1.21.0-i686-unknown-haiku"
+		;;
+	x86_64)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rust-$portVersion-x86_64-unknown-haiku.tar.xz"
+CHECKSUM_SHA256="cbb7f705398b76c5e2dd18119cdb9af2c042720449479b482a39309fb270ded8"
+SOURCE_DIR="rust-$portVersion-x86_64-unknown-haiku"
+		;;
+	*)
+SOURCE_URI="http://dl.rust-on-haiku.com/dist/$portVersion/rustc-$portVersion-src.tar.xz"
+CHECKSUM_SHA256="7741c2f3652813b7ecd0233305a5ca0de561fea2849049651a79521d2c437f0f"
+SOURCE_DIR="rustc-$portVersion-src"
+		;;
+esac
+
+DISABLE_SOURCE_PACKAGE="true"
+
+cargoVersion="0.22.0"
+rlsVersion="0.121.0"
+
+PROVIDES="
+	rust$secondaryArchSuffix = $portVersion
```

Comment:
Hmm, I wonder if we should not change this to:
```diff
 PROVIDES="
-	rust$secondaryArchSuffix = $portVersion
+	rust${secondaryArchSuffix}_bin = $portVersion
```
and add a
```
CONFLICTS="
	rust$secondaryArchSuffix
	"
```

I think this was discussed but I don't remember what was the conclusion, so I might very well be missing something...

I also wonder if we should not define a `commandBinDir` and `commandSuffix` to make the recipe `x86_64-hybrid`-_ready_.
(We would append a `commandSuffix` to `cmd:rust{c,doc,_lldb}`.)

Reply:
The outcome was that the recipe now defines a REPLACES line, which forces the uninstall of earlier non_bin recipes. About being 64 bit ready: I foresee some problems with hardcoded paths. I think it is better to look at the situation then and solve the problems then. Rust releases every six weeks, so there will be opportunity in the future.

I have made the smaller changes. Thanks for reviewing!

Reply:
I think the clock of your VM is not correctly set. I suspect that:
1) your real timezone is UTC+x (i.e. your latitude is close to 15*x East)
2) the clock of your VM is set to UTC and your localtime is being used as the UTC time.

This is not very important, but there is a little side effect: GitHub shows your commit updates below the last message, unless that message was added at least x hours after the last commit update.

For our visitors, the following commands can help finding the right settings:
```
date
TZ=UTC date
```